llvm/lib/Transforms/Scalar/Reassociate.cpp

```diff
@@ -648,6 +651,13 @@ static bool LinearizeExprTree(Instruction *I,
     // Ensure the leaf is only output once.
     It->second = 0;
     Ops.push_back(std::make_pair(V, Weight));
+    if (Opcode == Instruction::Add && Flags.AllKnownNonNegative &&
+        Flags.HasNSW) {
+      // Note: AllKnownNegative can be true in a case where one of the operands
+      // is negative, but one the operators is not NSW. AllKnownNegative should
+      // not be used independently of HasNSW
```

Comment:
This comment should probably be on the member in the struct declaration?

Reply:
Makes sense, done.